pkg/restapi/operation/openapi.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package operation
+
+import "github.com/trustbloc/edv/pkg/restapi/models"
+
+// genericError model
+//
+// swagger:response genericError
+type genericError struct { // nolint: unused,deadcode
+	// in: body
+	ErrMsg string
+}
+
+// createVaultReq model
+//
+// swagger:parameters createVaultReq
+type createVaultReq struct { // nolint: unused,deadcode
+	// in: body
+	NewVaultRequest models.DataVaultConfiguration
+}
+
+// createVaultRes model
+//
+// swagger:response createVaultRes
+type createVaultRes struct { // nolint: unused,deadcode
+	// For some reason this doesn't display in the Swagger response UI currently...
+	// To see the returned location, you'll have to open the developer console.
+	// The location will be in the form of HostURL/encrypted-data-vaults/vaultID
+	Location string
+}
+
+// queryVaultReq model
+//
+// swagger:parameters queryVaultReq
+type queryVaultReq struct { // nolint: unused,deadcode
+	// in: path
+	// required: true
+	VaultID string `json:"vaultID"`
+	// in: body
+	QueryRequest models.Query
+}
+
+// queryVaultRes model
+//
+// swagger:response queryVaultRes
+type queryVaultRes struct { // nolint: unused,deadcode
+	// in: body
+	QueryResults string
+}
+
+// createDocumentReq model
+//
+// swagger:parameters createDocumentReq
+type createDocumentReq struct { // nolint: unused,deadcode
+	// in: path
+	// required: true
+	VaultID string `json:"vaultID"`
+	// in: body
+	NewDocument models.StructuredDocument
+}
+
+// createDocumentRes model
+//
+// swagger:response createDocumentRes
+type createDocumentRes struct { // nolint: unused,deadcode
+	// For some reason this doesn't display in the Swagger response UI currently...
+	// To see the returned location, you'll have to open the developer console.
+	// The location will be in the form of HostURL/encrypted-data-vaults/vaultID/documents/docID
+	Location string
+}
+
+// readDocumentReq model
+//
+// swagger:parameters readDocumentReq
+type readDocumentReq struct { // nolint: unused,deadcode
+	// in: path
+	// required: true
+	VaultID string `json:"vaultID"`
+	// in: path
+	// required: true
+	DocID string `json:"docID"`
+}
+
+// readDocumentRes model
+//
+// swagger:response readDocumentRes
+type readDocumentRes struct { // nolint: unused,deadcode
+	// in: body
+	RetrievedDocument string
+}
+
+// changeLogSpecReq model
+//
+// swagger:parameters changeLogSpecReq
+type changeLogSpecReq struct { // nolint: unused,deadcode
+	// in: body
+	Body struct {
+		// The new log specification
+		//
+		// Required: true
+		// Example: restapi=debug:edv-rest=critical:error
+		Spec string `json:"spec"`
+	}
+}
+
+// changeLogSpecRes model
+//
+// swagger:response changeLogSpecRes
+type changeLogSpecRes struct { // nolint: unused,deadcode
```

Comment:
why empty models? if no response then don't specify response models.

Reply:
There's a couple of things going on here...
1. The spec generator doesn't allow me to not specify a response model.
2. The reason this looks so weird is because the response isn't JSON. I can't find any syntax to mark it as a non-JSON response. This will be addressed in #90.

Reply:
https://github.com/hyperledger/aries-framework-go/blob/master/pkg/controller/rest/messaging/operation.go#L121